realtime-checkup: Add container-disk image create manual (#77409)
openshift/openshift-docs: modules/virt-building-real-time-container-disk-image.adoc

// * virt//support/monitoring/virt-running-cluster-checkups.adoc

:_mod-docs-content-type: PROCEDURE
[id="virt-building-realtime-container-disk-image_{context}"]

> = Building a realtime container disk image for {op-system-base} virtual machines

Reviewer: The term `real-time` is all over the documentation.
Please consider leaving it as-is.

> You can create a custom {op-system-base-full} 8 Operating system image in `QCOW2` format and use it to create a container disk image. You can store the container disk image in a registry that is accessible from your cluster and specify the image location in the `spec.param.vmUnderTestContainerDiskImage` attribute of the realtime checkup config map.

Reviewer: What is the reason to change the wording from the original text?
It is the same as in the DPDK checkup.

Author: it's not the same.
On the DPDK you have instruction on the VM where the image is created on, here we do not.

// * virt//support/monitoring/virt-running-cluster-checkups.adoc

:_mod-docs-content-type: PROCEDURE
[id="virt-building-realtime-container-disk-image_{context}"]
= Building a container disk image for {op-system-base} virtual machines

You can create a custom {op-system-base-full} 8 Operating system image in `QCOW2` format and use it to create a container disk image. You can store the container disk image in a registry that is accessible from your cluster and specify the image location in the `spec.param.vmUnderTestContainerDiskImage` attribute of the real-time checkup config map.

.Procedure

. Run a {op-system-base-full} 8 Operating system image and set the root password to "redhat"

. Login as root and install the following packages:

> - kernel-rt

Reviewer 2: Change syntax of the list items as follows:

```suggestion
** kernel-rt
```

Author: DONE

- tuned
- tuned-profiles-realtime
- tuned-profiles-nfv-guest
- rt-tests
- cloud-init
----
+
[NOTE]
====

> You may need to enable to following repositories on subscription-manager:

Reviewer 2: Same suggestion as earlier for list item syntax:

s/-/**

Author: DONE

> . Run a {op-system-base-full} 8 Operating system image create a new root user with the following user and password:

Reviewer: ```suggestion
. Create a new {op-system-base-full} 8 Operating Virtual Machine.
. Login to the newly created virtual machine using the virtctl utility.
. Set the `root` user's password to "redhat" which is what the checkup expects.
```

> name = "root"
> password = "redhat"

Reviewer: ```suggestion
```

- rhel-8-for-x86_64-rt-rpms
- rhel-8-for-x86_64-nfv-rpms
====

> . Disable the following services from running:

Reviewer 2: See my other comment about list item syntax

Author: DONE

> . Login to this root user and install the following packages:

Reviewer: ```suggestion
. Install the following packages:
```

// * virt//support/monitoring/virt-running-cluster-checkups.adoc

:_mod-docs-content-type: PROCEDURE
[id="virt-building-realtime-container-disk-image_{context}"]
= Building a realtime container disk image for {op-system-base} virtual machines

You can create a custom {op-system-base-full} 8 Operating system image in `QCOW2` format and use it to create a container disk image. You can store the container disk image in a registry that is accessible from your cluster and specify the image location in the `spec.param.vmUnderTestContainerDiskImage` attribute of the realtime checkup config map.

.Procedure

. Run a {op-system-base-full} 8 Operating system image create a new root user with the following user and password:
name = "root"
password = "redhat"

. login to this root user and install the following packages:
+
[source,terminal]
----

> # dnf install kernel-rt tuned tuned-profiles-realtime tuned-profiles-nfv-guest rt-tests cloud-init

Reviewer: In the guest OS configuration instructions, there is a step for enabling the real-time and NFV repos.
Please consider explicitly mentioning it or making the step more general by not showing the explicit command.

Author: ACK. DONE

----
+
[NOTE]
====
You may need to enable to following repositories:
[source,terminal]
----
# subscription-manager repos --enable rhel-8-for-x86_64-rt-rpms rhel-8-for-x86_64-nfv-rpms
----
====

. Disable the following services:
[source,terminal]
----
# systemctl disable NetworkManager-wait-online sshd irqbalance
----

. Copy this script on the image:
+
[source,terminal]
----
# cat <<EOF >customize-vm.sh
#!/bin/bash

set -xe

# Disable swap
swapoff -a
sed -i '/swap/s/^/#/' /etc/fstab

# Enable guest-exec,guest-exec-status on the qemu-guest-agent configuration
sed -i '/^BLACKLIST_RPC=/ { s/guest-exec-status//; s/guest-exec//g }' /etc/sysconfig/qemu-ga
sed -i '/^BLACKLIST_RPC=/ { s/,\+/,/g; s/^,\|,$//g }' /etc/sysconfig/qemu-ga
EOF
----

> . Run the script on the image:

Reviewer: I can image that there should be a step to export the VM to a qcow2 image file.
In that case, the steps performed by the script, could be performed inside the guest prior to exporting it, thus making the need to use virt-customize redundant.

Author: We're not running virt-customize anymore - I just kept the scipt name `customize-vm.sh`

Reviewer: I see.
Since this script requires `root` privileges please explicitly state it.

Author: Since the manual is all in root user (the one we created and logged in with on earlier steps) - I'm not sure it's needed. WDYT?
